Added Accessibility Guide For Terra-Image (#3592)
cerner/terra-core: packages/terra-core-docs/src/terra-dev-site/doc/image/AccessibilityGuide.3.doc.mdx

**Decorative images** must have a null (empty) alt text (alt="") so that they can be ignored by assistive technologies, such as screen readers. Text values for these types of images would add audible clutter to screen reader output or could distract users if the topic is different from that in adjacent text. Leaving out the alt attribute is also not an option because when it is not provided, some screen readers will announce the file name of the image instead.

_Whether to treat an image as informative or decorative is a judgment that should be made by Author, based on the reason for including the image on the page._

Following section describes how to use the `alt` attribute of the `Terra-Image` component in various situations.

> ### 1. Use [Image](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/image).

Reviewer: ```suggestion
### 1. Use [Image](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/image)
```

> - When the `Image` contribute meaning to the current page or context.

Reviewer: ```suggestion
- When the `Image` contributes meaning to the current page or context.
```

- When the `Image` is a simple graphic or photograph that describes the intended meaning of image.

> - When the `Image` is conveying an impression or emotion.

Reviewer: ```suggestion
- When the `Image` conveys an impression or emotion.
```

See [Informative Images](https://www.w3.org/WAI/tutorials/images/informative/) for more info.

#### Related or Additional WCAG Resources for Informative Image.
- [G94: Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content](https://www.w3.org/WAI/WCAG21/Techniques/general/G94).
- [H37: Using alt attributes on img elements](https://www.w3.org/WAI/WCAG21/Techniques/html/H37).

#### Examples
<InformativeImage />

> ### 2. Use [DecorativeImage](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/decorative-image).

Reviewer: ```suggestion
### 2. Use [DecorativeImage](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/image/decorative-image)
```

> - When `Image` contain text and the text is also present as real text nearby.
> - When `Image` contain text and the text is only shown for visual effects.
> - When the `Image` purely decorative or not intended for the user.

Reviewer: ```suggestion
- When `Image` contains text and the text is also present as real text nearby.
- When `Image` contains text and the text is only shown for visual effects.
- When `Image` is purely decorative or not intended for the user.
```

See [Decorative Images](https://www.w3.org/WAI/tutorials/images/decorative/) for more info.

#### Related or Additional WCAG Resources for Decorative Image.
- [H2: Combining adjacent image and text links for the same resource](https://www.w3.org/WAI/WCAG21/Techniques/html/H2).
- [H67: Using null alt text and no title attribute on img elements for images that AT should ignore](https://www.w3.org/WAI/WCAG21/Techniques/html/H67).

#### Examples
<DecorativeImageBar title="Decorative Image"/>

_Note: `DecorativeImage` component sets `alt` prop to empty `""` value internally to make `terra-image` as Decorative Image._

## Usability Expectations:

### Keyboard Navigation
 

> Terra Images does not have any keyboard navigation expectations on its own. Any content provided to the `startContent`, `endContent`, or `children` should follow the interaction expectations of those elements.

Reviewer: I see the term "Terra Images" sometimes being used (on line 12) as a plural and here as a singular. Is this referring to the component name (Terra Image) or just images in general?
